e2xgrader/apps/api.py

```diff
@@ -140,3 +144,187 @@ def get_task_submissions(self, assignment_id, notebook_id, task_id):
+
+    def update_cell_content(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        cell_content: string
+            The updated content of the cell
+        Returns
+        -------
+        checksum_id: str
+            Generates new checksum id after changes.
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
+        for i in nb.cells:
+            if i['metadata']['nbgrader']['grade_id'] == cell_id:
+                cell_content = i['source']
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, source = cell_content)
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, checksum = utils.compute_checksum(i))
+
+                return utils.compute_checksum(i)
+
+        return None
+
+    def update_cell(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        cell_content: string
+            The updated content of the cell
+        Returns
+        -------
+        checksum_id: str
+            Generates new checksum id after changes.
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
+        for i in nb.cells:
+            if i['metadata']['nbgrader']['grade_id'] == cell_id:
+                i['source'] = cell_content
+                fname = "source/" + assignment + "/" + notebook + ".ipynb"
+                with open(fname, 'w') as f:
+                    nbformat.write(nb, f)
+                
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, source = cell_content)
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, checksum = utils.compute_checksum(i))
+
+                return utils.compute_checksum(i)
+
+        return None
+
+    def student_autograde(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        Returns
+        -------
+        None
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
```

Comment:
Again use coursedir here to find correct path and os.path.join to join it

Reply:
This function was not required and hence deleted.